### PR TITLE
Remove unused player ID from PLAYER_STATUS message.

### PR DESCRIPTION
**Pull request by o01eg**
Breaks networking compatibility.

### CODE REVIEW

File: network/Message.cpp [hunk 1]
```diff
@@ -404,15 +404,13 @@ Message TurnProgressMessage(Message::TurnProgressPhase phase_id) {
     return Message(Message::TURN_PROGRESS, os.str());
 }
 
-Message PlayerStatusMessage(int about_player_id,
-                            Message::PlayerStatus player_status,
+Message PlayerStatusMessage(Message::PlayerStatus player_status,
```
Comment (geoffthemedio) [inline network/Message.cpp:407]: Swapping the order of these causes a lot of changes elsewhere. Is it necessary or beneficial?

**Comment by geoffthemedio** [comment]
Could probably restore network compatibility with some player ID empire ID lookups, but is thete much reason to bother with it?

File: network/Message.cpp (hunk 1 above)
Comment (o01eg) [reply network/Message.cpp:407]: It was player ID, now there empire ID and both have type int. So I decide to swap order to don't send player ID as empire ID somewhere.

**Comment by o01eg** [comment]
Current {player id, status, empire id} was compatible with {player id, status}. I decided to remove player ID part after #2420

**Comment by geoffthemedio** [comment]
Not sure what that means re: my question... Point was you could look up player or empire IDs from the other.

**Comment by o01eg** [comment]
Using player id for `PlayerStatus` is deprecated in https://github.com/freeorion/freeorion/pull/2352
This PR is a cleanup.

> Point was you could look up player or empire IDs from the other.

It possible to have players (observers and moderators) without empire and empire without connected players.

**Pull request merged**